app/assets/javascripts/admin/addon/controllers/admin-customize-themes-show.js

```diff
@@ -254,6 +254,18 @@ export default class AdminCustomizeThemesShowController extends Controller {
     return userId > 0 && !defaultTheme;
   }
 
+  @discourseComputed(
+    "model.colorPalette.colors.@each.{hex,originalHex,dark_hex,originalDarkHex}"
+  )
+  hasChangedColors() {
+    return this.model.colorPalette.colors.some((color) => {
+      return (
+        color.hex !== color.originalHex ||
+        color.dark_hex !== color.originalDarkHex
+      );
+    });
+  }
```

Comment:
Instead of having to inspect each object for changes, perhaps we can pass a tracked property to the `ColorPaletteEditor` component for it to update this tracked property when changes are made?

Reply:
I've reworked this bit so we no longer need to loop through the colors.